frame/transaction-payment/src/lib.rs

```diff
@@ -229,44 +230,84 @@ impl Default for Releases {
+	use super::*;
 
-	/// The fee to be paid for making a transaction; the per-byte portion.
-	type TransactionByteFee: Get<BalanceOf<Self>>;
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
 
-	/// Convert a weight value into a deductible fee based on the currency type.
-	type WeightToFee: WeightToFeePolynomial<Balance=BalanceOf<Self>>;
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Handler for withdrawing, refunding and depositing the transaction fee.
+		/// Transaction fees are withdrawn before the transaction is executed.
+		/// After the transaction was executed the transaction weight can be
+		/// adjusted, depending on the used resources by the transaction. If the
+		/// transaction weight is lower than expected, parts of the transaction fee
+		/// might be refunded. In the end the fees can be deposited.
+		type OnChargeTransaction: OnChargeTransaction<Self>;
 
-	/// Update the multiplier of the next block, based on the previous block's weight.
-	type FeeMultiplierUpdate: MultiplierUpdate;
-}
+		/// The fee to be paid for making a transaction; the per-byte portion.
+		#[pallet::constant]
+		type TransactionByteFee: Get<BalanceOf<Self>>;
 
-decl_storage! {
-	trait Store for Module<T: Config> as TransactionPayment {
-		pub NextFeeMultiplier get(fn next_fee_multiplier): Multiplier = Multiplier::saturating_from_integer(1);
+		/// Convert a weight value into a deductible fee based on the currency type.
+		type WeightToFee: WeightToFeePolynomial<Balance = BalanceOf<Self>>;
 
-		StorageVersion build(|_: &GenesisConfig| Releases::V2): Releases;
+		/// Update the multiplier of the next block, based on the previous block's weight.
+		type FeeMultiplierUpdate: MultiplierUpdate;
 	}
-}
-
-decl_module! {
-	pub struct Module<T: Config> for enum Call where origin: T::Origin {
-		/// The fee to be paid for making a transaction; the per-byte portion.
-		const TransactionByteFee: BalanceOf<T> = T::TransactionByteFee::get();
 
+	#[pallet::extra_constants]
+	impl<T: Config> Pallet<T> {
+		//TODO: rename to snake case after https://github.com/paritytech/substrate/issues/8826 fixed.
+		#[allow(non_snake_case)]
 		/// The polynomial that is applied in order to derive fee from weight.
-		const WeightToFee: Vec<WeightToFeeCoefficient<BalanceOf<T>>> =
-			T::WeightToFee::polynomial().to_vec();
+		fn WeightToFee() -> Vec<WeightToFeeCoefficient<BalanceOf<T>>> {
+			T::WeightToFee::polynomial().to_vec()
+		}
+	}
+
+	#[pallet::type_value]
+	pub fn NextFeeMultiplierOnEmpty() -> Multiplier { Multiplier::saturating_from_integer(1) }
+	#[pallet::storage]
+	#[pallet::getter(fn next_fee_multiplier)]
+	pub type NextFeeMultiplier<T: Config> = StorageValue<
+		_,
+		Multiplier,
+		ValueQuery,
+		NextFeeMultiplierOnEmpty
+	>;
+
+	#[pallet::storage]
+	pub(super) type StorageVersion<T: Config> = StorageValue<_, Releases, ValueQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T> {
```

Comment:
I removed the generic here as it is not needed

(but also I think we should remove the GenesisConfig completly and remove the `Releases` storage. If people wonder about which version their pallet is, then they should look at the pallet version stored.
Note that substrate node sand polkadot/kusama nodes don't run this genesis config (it is not configured in construct_runtime).
But we can do in a follow-up PR)